src/features/vpn/services.ts

```diff
@@ -13,44 +13,39 @@ import { checkDeviceExists } from '../device-state/middleware';
 
 const { api } = sbvrUtils;
 
-const checkAuth = (() => {
-	const authQuery = _.once(() =>
-		api.resin.prepare<{ uuid: string }>({
-			resource: 'device',
-			id: {
-				uuid: { '@': 'uuid' },
-			},
-			options: {
-				$select: 'id',
-			},
-		}),
-	);
-	return multiCacheMemoizee(
-		async (uuid: string, req: permissions.PermissionReq): Promise<number> => {
-			try {
-				const device = await authQuery()({ uuid }, undefined, { req });
-				// for now if the api key is able to read the device then it has vpn access
-				return device != null ? 200 : 403;
-			} catch (err) {
-				if (err instanceof errors.UnauthorizedError) {
-					// We handle `UnauthorizedError` specially so that we can cache it as it's a relatively
-					// common case for devices that have been deleted but are still online/trying to connect
-					return 401;
-				}
-				throw err;
+const checkAuth = multiCacheMemoizee(
+	async (uuid: string, req: permissions.PermissionReq): Promise<number> => {
+		try {
+			const deviceResult = (await api.resin.post({
+				passthrough: {
+					req,
+				},
+				url: `device(uuid='${uuid}')/canAccess`,
+				body: { action: 'cloudlink' },
+			})) as { d?: Array<{ id: number }> };
+			const device = deviceResult.d?.[0];
+			// Note: /canAccess atm always throws a 401 if it doesn't find a device.
+			// TODO: Rethink whether we need to make a separate request to return a 403.
```

Comment:
The VPN seems to be treating them the same atm and just returns a 401 to the device.